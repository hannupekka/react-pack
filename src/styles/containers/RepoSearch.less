@import '~@app/styles/main';

.searc {
  img {
    display: block;
    margin-top: @padding;
  }
}

.input {
  composes: base__input;
  margin: 0 @padding;
}

.button {
  composes: base__button;

  & + & {
    margin-left: @padding;
  }
}

.list {
  margin-top: @padding;

  a {
    display: block;
  }
}
