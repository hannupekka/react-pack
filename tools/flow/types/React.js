declare type ElementType = React$Element<any>
