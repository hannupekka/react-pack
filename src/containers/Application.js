// @flow
import styles from 'styles/containers/Application.less';
import React, { Component } from 'react';
import CSSModules from 'react-css-modules';
import Header from 'components/Header';

type Props = {
  children: ElementType | Array<ElementType>
}

// eslint-disable-next-line react/prefer-stateless-function
class Application extends Component {
  props: Props;
  render(): ElementType {
    const { children } = this.props;

    return (
      <div>
        <Header />
        <section styleName="content">
          {children}
        </section>
      </div>
    );
  }
}

export default CSSModules(Application, styles);
