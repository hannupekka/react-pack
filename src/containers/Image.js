import styles from 'styles/image.less';
import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import ImmutablePropTypes from 'react-immutable-proptypes';
import CSSModules from 'react-css-modules';
import * as ImageActions from 'actions/image';
import Loader from 'components/loader';
import Error from 'components/Error';

class Image extends Component {
  render() {
    const { isLoading, isError, image, actions } = this.props;

    if (isLoading) {
      return <Loader />;
    }

    if (isError) {
      return <Error message="Error requesting image!" />;
    }

    return (
      <div styleName="image">
        <button onClick={() => actions.requestImage()}>Get random image</button>
        {image.get('image_url') && <img src={image.get('image_url')} />}
      </div>
    );
  }
}

Image.propTypes = {
  actions: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
  image: ImmutablePropTypes.map,
  params: PropTypes.object // Inserted by react-router-redux
};

const select = store => {
  return {
    isLoading: store.image.get('isLoading'),
    isError: store.image.get('isError'),
    image: store.image.get('image')
  };
};

const mapActions = dispatch => {
  return {
    actions: bindActionCreators(ImageActions, dispatch)
  };
};

export default connect(
  select,
  mapActions
)(CSSModules(Image, styles));
