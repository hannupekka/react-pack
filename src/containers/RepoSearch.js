// @flow
import styles from 'styles/containers/RepoSearch.less';
import React, { Component } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Map } from 'immutable';
import { connect } from 'react-redux';
import CSSModules from 'react-css-modules';
import { pure } from 'recompose';
import * as reposActions from 'redux/modules/repos';
import getVisibleRepos from 'redux/selectors/repos';
import Loader from 'components/Loader';
import Error from 'components/Error';
import Repo from 'components/Repo';

type Props = {
  repos: ImmutablePropTypes.map.isRequired,
  users: ImmutablePropTypes.map.isRequired,
  isError: boolean,
  isLoading: boolean,
  showForks: boolean,
  dispatch: Function
};

export class RepoSearch extends Component {
  props: Props;

  bindUsername: Function;
  username: HTMLInputElement;

  constructor(props: Object) {
    super(props);

    this.bindUsername = c => (this.username = c);
  }

  onFetchRepos = (): void => {
    const { dispatch } = this.props;
    const username: string = this.username.value;
    dispatch(reposActions.fetchRepos(username));
  }

  onToggleShowForks = (): void => {
    const { dispatch } = this.props;
    dispatch(reposActions.toggleShowForks());
  }

  maybeRenderFilter = (): ?React$Element<any> => {
    const { repos, showForks } = this.props;

    if (repos.size === 0) {
      return null;
    }

    const text = showForks ? 'Hide forks' : 'Include forks';

    return (
      <button styleName="button" onClick={this.onToggleShowForks}>
        {text} <i className="fa fa-filter"></i>
      </button>
    );
  }

  maybeRenderRepoList = (): ?React$Element<any> => {
    const { repos, users } = this.props;

    if (repos.size === 0) {
      return null;
    }

    return repos.map((repo) => {
      const id = repo.get('id');

      const params = {
        name: repo.get('name'),
        url: repo.get('html_url'),
        user: users.getIn([repo.get('owner').toString(), 'login'])
      };

      return <Repo key={id} {...params} />;
    }).toArray();
  }

  render(): React$Element<any> {
    const { isLoading, isError } = this.props;

    return (
      <div styleName="repo-search">
        { isError && <Error message="Repositories not found!" />}
        <label htmlFor="search">Username</label>
        <input
          type="text"
          styleName="input"
          defaultValue="hannupekka"
          ref={this.bindUsername}
        />
        <button styleName="button" onClick={this.onFetchRepos}>
          Get users repositories <i className="fa fa-search"></i>
        </button>
        {this.maybeRenderFilter()}
        {isLoading && <Loader />}
        <div styleName="repo-list">
          {this.maybeRenderRepoList()}
        </div>
      </div>
    );
  }
}

type Store = {
  +repos: Map<string, any>
};

const mapState: Object = (store: Store) => ({
  repos: getVisibleRepos(store),
  users: store.repos.getIn(['entities', 'users']),
  isError: store.repos.get('isError'),
  isLoading: store.repos.get('isLoading'),
  showForks: store.repos.get('showForks')
});

export default connect(
  mapState
)(pure(CSSModules(RepoSearch, styles)));
