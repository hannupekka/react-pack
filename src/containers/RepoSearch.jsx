import styles from 'styles/containers/RepoSearch.less';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import CSSModules from 'react-css-modules';
import map from 'lodash/map';
import { createStructuredSelector } from 'reselect';
import { hot } from 'react-hot-loader';
import * as reposActions from 'redux/modules/repo';
import {
  getVisibleRepos,
  getUsers,
  getIsError,
  getIsLoading,
  getShowForks,
} from 'redux/modules/repo/selectors';
import Loader from 'components/Loader';
import Error from 'components/Error';
import Repo from 'components/Repo';

export class RepoSearch extends PureComponent {
  constructor(props) {
    super(props);

    this.bindUsername = c => {
      this.username = c;
    };
  }

  onFetchRepos = () => {
    const { dispatch } = this.props;
    const username = this.username.value;
    dispatch(reposActions.fetchRepos(username));
  };

  onToggleShowForks = () => {
    const { dispatch } = this.props;
    dispatch(reposActions.toggleShowForks());
  };

  maybeRenderFilter = () => {
    const { repos, showForks } = this.props;

    if (Object.keys(repos).length === 0) {
      return null;
    }

    const text = showForks ? 'Hide forks' : 'Include forks';

    return (
      <button styleName="button" onClick={this.onToggleShowForks}>
        {text} <i className="fa fa-filter" />
      </button>
    );
  };

  maybeRenderRepoList = () => {
    const { repos, users } = this.props;

    if (Object.keys(repos).length === 0) {
      return null;
    }

    return map(repos, repo => {
      const { id, name, html_url: url, owner } = repo;

      const params = {
        name,
        url,
        user: users[owner].login,
      };

      return <Repo key={id} {...params} />;
    });
  };

  render() {
    const { isLoading, isError } = this.props;

    return (
      <div styleName="repo-search">
        {isError && <Error message="Repositories not found!" />}
        <label htmlFor="search">
          Username
          <input type="text" styleName="input" defaultValue="hannupekka" ref={this.bindUsername} />
        </label>
        <button styleName="button" onClick={this.onFetchRepos}>
          Get users repositories <i className="fa fa-search" />
        </button>
        {this.maybeRenderFilter()}
        {isLoading && <Loader />}
        <div styleName="repo-list">{this.maybeRenderRepoList()}</div>
      </div>
    );
  }
}

RepoSearch.propTypes = {
  repos: PropTypes.shape().isRequired,
  users: PropTypes.shape().isRequired,
  isError: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  showForks: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapState = createStructuredSelector({
  repos: getVisibleRepos,
  users: getUsers,
  isError: getIsError,
  isLoading: getIsLoading,
  showForks: getShowForks,
});

export default hot(module)(connect(mapState)(CSSModules(RepoSearch, styles)));
