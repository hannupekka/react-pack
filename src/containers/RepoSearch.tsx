import styles from '@app/styles/containers/RepoSearch.less';
import * as React from 'react';
import { connect } from 'react-redux';
import map from 'lodash-es/map';
import { pure } from 'recompose';
import { createStructuredSelector } from 'reselect';
import { TDispatch, IRepo, IUser } from '@app/types';
import * as reposActions from '@app/redux/repos';
import {
  getVisibleRepos,
  getUsers,
  getIsError,
  getIsLoading,
  getShowForks,
} from '@app/redux/repos/selectors';
import Loader from '@app/components/Loader';
import Error from '@app/components/Error';
import Repo from '@app/components/Repo';

interface Props {
  dispatch: TDispatch;
  repos: {
    [id: string]: IRepo;
  };
  users: {
    [id: string]: IUser;
  };
  showForks: boolean;
  isLoading: boolean;
  isError: boolean;
}

export class RepoSearch extends React.Component<Props> {
  bindUsername: (c: HTMLInputElement) => void;
  username: HTMLInputElement;

  constructor(props: Props) {
    super(props);

    this.bindUsername = c => {
      this.username = c;
    };
  }

  onFetchRepos = () => {
    const { dispatch } = this.props;
    const username = this.username.value;
    dispatch(reposActions.fetchRepos(username));
  };

  onToggleShowForks = () => {
    const { dispatch } = this.props;
    dispatch(reposActions.toggleShowForks());
  };

  maybeRenderFilter = () => {
    const { repos, showForks } = this.props;

    if (Object.keys(repos).length === 0) {
      return null;
    }

    const text = showForks ? 'Hide forks' : 'Include forks';

    return (
      <button className={styles.button} onClick={this.onToggleShowForks}>
        {text} <i className="fa fa-filter" />
      </button>
    );
  };

  maybeRenderRepoList = () => {
    const { repos, users } = this.props;

    if (Object.keys(repos).length === 0) {
      return null;
    }

    return map(repos, (repo: IRepo) => {
      const { id, name, html_url: url, owner } = repo;

      const params = {
        name,
        url,
        user: users[owner].login,
      };

      return <Repo key={id} {...params} />;
    });
  };

  render(): JSX.Element {
    const { isLoading, isError } = this.props;

    return (
      <div className={styles.search}>
        {isError && <Error message="Repositories not found!" />}
        <label htmlFor="search">Username</label>
        <input
          type="text"
          className={styles.input}
          defaultValue="hannupekka"
          ref={this.bindUsername}
        />
        <button className={styles.button} onClick={this.onFetchRepos}>
          Get users repositories <i className="fa fa-search" />
        </button>
        {this.maybeRenderFilter()}
        {isLoading && <Loader />}
        <div className={styles.list}>{this.maybeRenderRepoList()}</div>
      </div>
    );
  }
}

const mapState = createStructuredSelector({
  repos: getVisibleRepos,
  users: getUsers,
  isError: getIsError,
  isLoading: getIsLoading,
  showForks: getShowForks,
});

export default connect(mapState)(pure(RepoSearch));
