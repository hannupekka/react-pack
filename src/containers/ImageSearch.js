// @flow

import styles from 'styles/containers/ImageSearch';
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import ImmutablePropTypes from 'react-immutable-proptypes';
import CSSModules from 'react-css-modules';
import { onlyUpdateForKeys } from 'recompose';
import * as imageActions from 'redux/modules/image';
import Loader from 'components/Loader';
import Error from 'components/Error';
import Image from 'components/Image';

class ImageSearch extends Component {
  constructor(props) {
    super(props);

    (this:any).bindSearch = (c) => ((this:any).search = c);
  }

  onFetchImage = () => {
    const search = (this:any).search.value;
    this.props.fetchImage(search);
  }

  render() {
    const { isLoading, isError, image } = this.props;

    if (isError) {
      return <Error message="Error requesting image!" />;
    }

    return (
      <div styleName="image-search">
        <label htmlFor="search">Search word</label>
        <input type="text" ref={(this:any).bindSearch} />
        <button onClick={this.onFetchImage}>
          Get random image <i className="fa fa-search"></i>
        </button>
        {isLoading && <Loader />}
        <Image src={image.get('fixed_height_downsampled_url')} />
      </div>
    );
  }
}

ImageSearch.propTypes = {
  fetchImage: PropTypes.func.isRequired,
  params: PropTypes.object,
  image: ImmutablePropTypes.map.isRequired,
  isError: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const select = store => ({
  image: store.image.get('image'),
  isError: store.image.get('isError'),
  isLoading: store.image.get('isLoading')
});

const mapActions = dispatch => {
  return {
    fetchImage: tag => dispatch(imageActions.fetchImage(tag))
  };
};

export default connect(
  select,
  mapActions
)(onlyUpdateForKeys(['isLoading', 'isError', 'image'])(CSSModules(ImageSearch, styles)));
