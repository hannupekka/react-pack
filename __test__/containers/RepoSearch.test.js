import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { fromJS, Map } from 'immutable';
import { RepoSearch } from 'containers/RepoSearch';

const REPOS = fromJS({
  "12482715": {
    "hooks_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/hooks",
    "default_branch": "master",
    "branches_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/branches{/branch}",
    "size": 144,
    "forks_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/forks",
    "forks_count": 0,
    "fork": false,
    "git_url": "git://github.com/hannupekka/fressi-gym-activity.git",
    "private": false,
    "keys_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/keys{/key_id}",
    "comments_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/comments{/number}",
    "trees_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/trees{/sha}",
    "has_downloads": true,
    "blobs_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/blobs{/sha}",
    "teams_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/teams",
    "watchers_count": 0,
    "stargazers_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/stargazers",
    "stargazers_count": 0,
    "contributors_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/contributors",
    "tags_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/tags",
    "issue_comment_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/comments{/number}",
    "events_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/events",
    "created_at": "2013-08-30T09:17:39Z",
    "issues_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues{/number}",
    "contents_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/contents/{+path}",
    "merges_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/merges",
    "name": "fressi-gym-activity",
    "languages_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/languages",
    "commits_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/commits{/sha}",
    "subscription_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/subscription",
    "clone_url": "https://github.com/hannupekka/fressi-gym-activity.git",
    "homepage": null,
    "issue_events_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/events{/number}",
    "mirror_url": null,
    "labels_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/labels{/name}",
    "url": "https://api.github.com/repos/hannupekka/fressi-gym-activity",
    "open_issues": 0,
    "statuses_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/statuses/{sha}",
    "forks": 0,
    "archive_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/{archive_format}{/ref}",
    "milestones_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/milestones{/number}",
    "owner": 4472477,
    "assignees_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/assignees{/user}",
    "has_wiki": true,
    "compare_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/compare/{base}...{head}",
    "git_tags_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/tags{/sha}",
    "updated_at": "2013-10-31T15:13:19Z",
    "watchers": 0,
    "notifications_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/notifications{?since,all,participating}",
    "ssh_url": "git@github.com:hannupekka/fressi-gym-activity.git",
    "language": "Python",
    "pushed_at": "2013-10-31T15:13:15Z",
    "downloads_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/downloads",
    "subscribers_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/subscribers",
    "id": 12482715,
    "svn_url": "https://github.com/hannupekka/fressi-gym-activity",
    "full_name": "hannupekka/fressi-gym-activity",
    "html_url": "https://github.com/hannupekka/fressi-gym-activity",
    "description": "Fetches, parses and formats gym activity for Fressi gyms",
    "releases_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/releases{/id}",
    "git_refs_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/refs{/sha}",
    "collaborators_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/collaborators{/collaborator}",
    "pulls_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/pulls{/number}",
    "deployments_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/deployments",
    "has_projects": true,
    "has_pages": false,
    "open_issues_count": 0,
    "git_commits_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/commits{/sha}",
    "has_issues": true
  },
  "13828937": {
    "hooks_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/hooks",
    "default_branch": "master",
    "branches_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/branches{/branch}",
    "size": 148,
    "forks_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/forks",
    "forks_count": 0,
    "fork": false,
    "git_url": "git://github.com/hannupekka/pyfibot-modules.git",
    "private": false,
    "keys_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/keys{/key_id}",
    "comments_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/comments{/number}",
    "trees_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/git/trees{/sha}",
    "has_downloads": true,
    "blobs_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/git/blobs{/sha}",
    "teams_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/teams",
    "watchers_count": 0,
    "stargazers_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/stargazers",
    "stargazers_count": 0,
    "contributors_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/contributors",
    "tags_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/tags",
    "issue_comment_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/issues/comments{/number}",
    "events_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/events",
    "created_at": "2013-10-24T10:15:26Z",
    "issues_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/issues{/number}",
    "contents_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/contents/{+path}",
    "merges_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/merges",
    "name": "pyfibot-modules",
    "languages_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/languages",
    "commits_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/commits{/sha}",
    "subscription_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/subscription",
    "clone_url": "https://github.com/hannupekka/pyfibot-modules.git",
    "homepage": "",
    "issue_events_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/issues/events{/number}",
    "mirror_url": null,
    "labels_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/labels{/name}",
    "url": "https://api.github.com/repos/hannupekka/pyfibot-modules",
    "open_issues": 0,
    "statuses_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/statuses/{sha}",
    "forks": 0,
    "archive_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/{archive_format}{/ref}",
    "milestones_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/milestones{/number}",
    "owner": 4472477,
    "assignees_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/assignees{/user}",
    "has_wiki": true,
    "compare_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/compare/{base}...{head}",
    "git_tags_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/git/tags{/sha}",
    "updated_at": "2014-01-08T09:21:00Z",
    "watchers": 0,
    "notifications_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/notifications{?since,all,participating}",
    "ssh_url": "git@github.com:hannupekka/pyfibot-modules.git",
    "language": "Python",
    "pushed_at": "2014-01-08T09:20:59Z",
    "downloads_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/downloads",
    "subscribers_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/subscribers",
    "id": 13828937,
    "svn_url": "https://github.com/hannupekka/pyfibot-modules",
    "full_name": "hannupekka/pyfibot-modules",
    "html_url": "https://github.com/hannupekka/pyfibot-modules",
    "description": "My collection of modules for Pyfibot the Python IRC bot",
    "releases_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/releases{/id}",
    "git_refs_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/git/refs{/sha}",
    "collaborators_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/collaborators{/collaborator}",
    "pulls_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/pulls{/number}",
    "deployments_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/deployments",
    "has_projects": true,
    "has_pages": false,
    "open_issues_count": 0,
    "git_commits_url": "https://api.github.com/repos/hannupekka/pyfibot-modules/git/commits{/sha}",
    "has_issues": true
  }
});

const USERS = fromJS({
  "4472477": {
    "gists_url": "https://api.github.com/users/hannupekka/gists{/gist_id}",
    "following_url": "https://api.github.com/users/hannupekka/following{/other_user}",
    "followers_url": "https://api.github.com/users/hannupekka/followers",
    "subscriptions_url": "https://api.github.com/users/hannupekka/subscriptions",
    "received_events_url": "https://api.github.com/users/hannupekka/received_events",
    "events_url": "https://api.github.com/users/hannupekka/events{/privacy}",
    "avatar_url": "https://avatars3.githubusercontent.com/u/4472477?v=3",
    "login": "hannupekka",
    "url": "https://api.github.com/users/hannupekka",
    "starred_url": "https://api.github.com/users/hannupekka/starred{/owner}{/repo}",
    "organizations_url": "https://api.github.com/users/hannupekka/orgs",
    "repos_url": "https://api.github.com/users/hannupekka/repos",
    "gravatar_id": "",
    "site_admin": false,
    "type": "User",
    "id": 4472477,
    "html_url": "https://github.com/hannupekka"
  }
});

describe('RepoSearch', () => {
  it('renders correctly with empty data', () => {
    const tree = renderer.create(
      <RepoSearch
        repos={Map()}
        users={Map()}
        isError={false}
        isLoading={false}
        dispatch={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with error', () => {
    const tree = renderer.create(
      <RepoSearch
        repos={Map()}
        users={Map()}
        isError
        isLoading={false}
        dispatch={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with loader', () => {
    const tree = renderer.create(
      <RepoSearch
        repos={Map()}
        users={Map()}
        isError={false}
        isLoading
        dispatch={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with data', () => {
    const tree = renderer.create(
      <RepoSearch
        repos={REPOS}
        users={USERS}
        isError={false}
        isLoading={false}
        dispatch={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
