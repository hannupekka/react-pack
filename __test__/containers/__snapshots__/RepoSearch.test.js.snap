// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoSearch renders correctly with all repos 1`] = `
<div
  className="repo-search"
>
  <label
    htmlFor="search"
  >
    Username
  </label>
  <input
    className="input"
    defaultValue="hannupekka"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Get users repositories 
    <i
      className="fa fa-search"
    />
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Hide forks
     
    <i
      className="fa fa-filter"
    />
  </button>
  <div
    className="repo-list"
  >
    <a
      href="https://github.com/hannupekka/fressi-gym-activity"
      rel="noopener noreferrer"
      target="_blank"
      title="author: hannupekka"
    >
      fressi-gym-activity
    </a>
    <a
      href="https://github.com/hannupekka/heroku-buildpack-python-libffi"
      rel="noopener noreferrer"
      target="_blank"
      title="author: hannupekka"
    >
      heroku-buildpack-python-libffi
    </a>
  </div>
</div>
`;

exports[`RepoSearch renders correctly with empty data 1`] = `
<div
  className="repo-search"
>
  <label
    htmlFor="search"
  >
    Username
  </label>
  <input
    className="input"
    defaultValue="hannupekka"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Get users repositories 
    <i
      className="fa fa-search"
    />
  </button>
  <div
    className="repo-list"
  />
</div>
`;

exports[`RepoSearch renders correctly with error 1`] = `
<div
  className="repo-search"
>
  <div
    className="error"
  >
    Repositories not found!
  </div>
  <label
    htmlFor="search"
  >
    Username
  </label>
  <input
    className="input"
    defaultValue="hannupekka"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Get users repositories 
    <i
      className="fa fa-search"
    />
  </button>
  <div
    className="repo-list"
  />
</div>
`;

exports[`RepoSearch renders correctly with forks hidden 1`] = `
<div
  className="repo-search"
>
  <label
    htmlFor="search"
  >
    Username
  </label>
  <input
    className="input"
    defaultValue="hannupekka"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Get users repositories 
    <i
      className="fa fa-search"
    />
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Include forks
     
    <i
      className="fa fa-filter"
    />
  </button>
  <div
    className="repo-list"
  >
    <a
      href="https://github.com/hannupekka/fressi-gym-activity"
      rel="noopener noreferrer"
      target="_blank"
      title="author: hannupekka"
    >
      fressi-gym-activity
    </a>
  </div>
</div>
`;

exports[`RepoSearch renders correctly with loader 1`] = `
<div
  className="repo-search"
>
  <label
    htmlFor="search"
  >
    Username
  </label>
  <input
    className="input"
    defaultValue="hannupekka"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Get users repositories 
    <i
      className="fa fa-search"
    />
  </button>
  <div>
    <div
      className="spinner"
    />
    <div
      className="text"
    >
      Loading
    </div>
  </div>
  <div
    className="repo-list"
  />
</div>
`;
